### ISSUE TITLE
"Must call rewindTo before calling getSourcePosition for an earlier line" with backslash followed by space in string continuation

### ISSUE THREAD

**Issue by kdkd** [google/closure-compiler#3492]
In the middle of a multiline string continuation, the sequence "\ " will produce a compiler exception.

```
function x() {
        a = "\
        \ \
        ";
};
```
```
java -jar closure-compiler-v20190929.jar --version
Closure Compiler (http://github.com/google/closure-compiler)
Version: v20190929
Built on: 2019-10-01 19:18
```
```
java -jar closure-compiler-v20190929.jar --js x.js
java.lang.RuntimeException: Exception parsing "x.js"
	at com.google.javascript.jscomp.parsing.ParserRunner.parse(ParserRunner.java:155)
	at com.google.javascript.jscomp.JsAst.parse(JsAst.java:153)
	at com.google.javascript.jscomp.JsAst.getAstRoot(JsAst.java:56)
	at com.google.javascript.jscomp.CompilerInput.getAstRoot(CompilerInput.java:134)
	at com.google.javascript.jscomp.Compiler.findModulesFromInput(Compiler.java:1890)
	at com.google.javascript.jscomp.Compiler.findModulesFromEntryPoints(Compiler.java:1857)
	at com.google.javascript.jscomp.Compiler.parseInputs(Compiler.java:1667)
	at com.google.javascript.jscomp.Compiler.parseForCompilationInternal(Compiler.java:933)
	at com.google.javascript.jscomp.Compiler.lambda$parseForCompilation$4(Compiler.java:916)
	at com.google.javascript.jscomp.CompilerExecutor$2.call(CompilerExecutor.java:102)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalArgumentException: Must call rewindTo before calling getSourcePosition for an earlier line (20 < 23)
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:300)
	at com.google.javascript.jscomp.parsing.parser.LineNumberScanner.getSourcePosition(LineNumberScanner.java:46)
	at com.google.javascript.jscomp.parsing.parser.LineNumberScanner.getSourceRange(LineNumberScanner.java:58)
	at com.google.javascript.jscomp.parsing.parser.Scanner.getTokenRange(Scanner.java:104)
	at com.google.javascript.jscomp.parsing.parser.Scanner.scanStringLiteral(Scanner.java:866)
	at com.google.javascript.jscomp.parsing.parser.Scanner.scanToken(Scanner.java:619)
	at com.google.javascript.jscomp.parsing.parser.Scanner.peekToken(Scanner.java:252)
	at com.google.javascript.jscomp.parsing.parser.Parser.peekToken(Parser.java:4228)
	at com.google.javascript.jscomp.parsing.parser.Parser.peekType(Parser.java:4214)
	at com.google.javascript.jscomp.parsing.parser.Parser.peek(Parser.java:4200)
	at com.google.javascript.jscomp.parsing.parser.Parser.peek(Parser.java:4192)
	at com.google.javascript.jscomp.parsing.parser.Parser.parseAssignment(Parser.java:2968)
	at com.google.javascript.jscomp.parsing.parser.Parser.parseAssignment(Parser.java:2992)
	at com.google.javascript.jscomp.parsing.parser.Parser.parse(Parser.java:2945)
	at com.google.javascript.jscomp.parsing.parser.Parser.parseExpression(Parser.java:2898)
	at com.google.javascript.jscomp.parsing.parser.Parser.parseExpressionStatement(Parser.java:1862)
	at com.google.javascript.jscomp.parsing.parser.Parser.parseStatementStandard(Parser.java:1679)
	at com.google.javascript.jscomp.parsing.parser.Parser.parseSourceElement(Parser.java:1157)
	at com.google.javascript.jscomp.parsing.parser.Parser.parseSourceElementList(Parser.java:1617)
	at com.google.javascript.jscomp.parsing.parser.Parser.parseFunctionBody(Parser.java:1608)
	at com.google.javascript.jscomp.parsing.parser.Parser.parseFunctionTail(Parser.java:1114)
	at com.google.javascript.jscomp.parsing.parser.Parser.parseFunctionDeclaration(Parser.java:1272)
	at com.google.javascript.jscomp.parsing.parser.Parser.parseSourceElement(Parser.java:1143)
	at com.google.javascript.jscomp.parsing.parser.Parser.parseScriptElement(Parser.java:382)
	at com.google.javascript.jscomp.parsing.parser.Parser.parseGlobalSourceElements(Parser.java:320)
	at com.google.javascript.jscomp.parsing.parser.Parser.parseProgram(Parser.java:306)
	at com.google.javascript.jscomp.parsing.ParserRunner.parse(ParserRunner.java:138)
	... 13 more
```

**Comment by kjin** [comment]
Created Google internal issue http://b/143471018

**Comment by thheller** [comment]
This exception also appears when trying to process https://unpkg.com/prettier@2.0.5/index.js.

```
mkdir test
cd test
npm init -y
npm install google-closure-compiler
curl -O https://unpkg.com/prettier@2.0.5/index.js
npx google-closure-compiler --js index.js
```
results in
```
index.js:34320: ERROR - [JSC_PARSE_ERROR] Parse error. '[' expected
    static(patterns, options) {
          ^

1 error(s), 0 warning(s)
java.lang.RuntimeException: Exception parsing "index.js"
        at com.google.javascript.jscomp.parsing.ParserRunner.parse(ParserRunner.java:155)
        at com.google.javascript.jscomp.JsAst.parse(JsAst.java:152)
        at com.google.javascript.jscomp.JsAst.getAstRoot(JsAst.java:55)
        at com.google.javascript.jscomp.CompilerInput.getAstRoot(CompilerInput.java:133)
        at com.google.javascript.jscomp.Compiler.findModulesFromInput(Compiler.java:1891)
        at com.google.javascript.jscomp.Compiler.findModulesFromEntryPoints(Compiler.java:1858)
        at com.google.javascript.jscomp.Compiler.parseInputs(Compiler.java:1667)
        at com.google.javascript.jscomp.Compiler.parseForCompilationInternal(Compiler.java:944)
        at com.google.javascript.jscomp.Compiler.lambda$parseForCompilation$4(Compiler.java:927)
        at com.google.javascript.jscomp.CompilerExecutor$2.call(CompilerExecutor.java:102)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
        at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:473)
        at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)
Caused by: java.lang.IllegalArgumentException: Must call rewindTo before calling getSourcePosition for an earlier line (1298797 < 1300077)
        at com.google.common.base.Preconditions.checkArgument(Preconditions.java:300)
        at com.google.javascript.jscomp.parsing.parser.LineNumberScanner.getSourcePosition(LineNumberScanner.java:46)
        at com.google.javascript.jscomp.parsing.parser.LineNumberScanner.getSourceRange(LineNumberScanner.java:58)
        at com.google.javascript.jscomp.parsing.parser.Scanner.getTokenRange(Scanner.java:100)
        at com.google.javascript.jscomp.parsing.parser.Scanner.nextTemplateLiteralTokenShared(Scanner.java:943)
        at com.google.javascript.jscomp.parsing.parser.Scanner.scanTemplateLiteral(Scanner.java:922)
        at com.google.javascript.jscomp.parsing.parser.Scanner.scanToken(Scanner.java:642)
        at com.google.javascript.jscomp.parsing.parser.Scanner.peekToken(Scanner.java:248)
        at com.google.javascript.jscomp.parsing.parser.Parser.peekToken(Parser.java:4442)
        at com.google.javascript.jscomp.parsing.parser.Parser.peekType(Parser.java:4428)
        at com.google.javascript.jscomp.parsing.parser.Parser.peek(Parser.java:4414)
        at com.google.javascript.jscomp.parsing.parser.Parser.peek(Parser.java:4406)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseArguments(Parser.java:3885)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseNewExpression(Parser.java:3835)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseLeftHandSideExpression(Parser.java:3611)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseUpdateExpression(Parser.java:3558)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseUnaryExpression(Parser.java:3516)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseExponentiationExpression(Parser.java:3486)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseMultiplicativeExpression(Parser.java:3464)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseAdditiveExpression(Parser.java:3442)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseShiftExpression(Parser.java:3419)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseRelational(Parser.java:3392)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseEquality(Parser.java:3368)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseBitwiseAND(Parser.java:3356)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseBitwiseXOR(Parser.java:3344)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseBitwiseOR(Parser.java:3332)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseLogicalAND(Parser.java:3320)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseLogicalOR(Parser.java:3308)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseShortCircuit(Parser.java:3278)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseConditional(Parser.java:3264)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseAssignment(Parser.java:2999)
        at com.google.javascript.jscomp.parsing.parser.Parser.parse(Parser.java:2970)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseExpression(Parser.java:2923)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseThrowStatement(Parser.java:2215)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseStatementStandard(Parser.java:1676)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseSourceElement(Parser.java:1158)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseScriptElement(Parser.java:383)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseGlobalSourceElements(Parser.java:321)
        at com.google.javascript.jscomp.parsing.parser.Parser.parseProgram(Parser.java:307)
        at com.google.javascript.jscomp.parsing.ParserRunner.parse(ParserRunner.java:138)
        ... 15 more
```
```

**Comment by brad4d** [comment]
https://tc39.es/ecma262/#sec-literals-string-literals

section of the spec defining string literals & line continuation

**Comment by concavelenz** [comment]
Fixed by 7278245 back in november
